core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1153,154 +1101,207 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
     finalLeaderIsrAndControllerEpoch
   }
 
-  class SessionExpirationListener() extends IZkStateListener with Logging {
-    this.logIdent = "[SessionExpirationListener on " + config.brokerId + "], "
+  private def checkAndTriggerPartitionRebalance(): Unit = {
```

Comment:
Haha I 100% agree that the method is hard to read as is the rest of KafkaController and kafka as a whole.

I want to minimize changes in this PR not related to moving the controller to a single-threaded event queue model. I'd be very willing to do a pure readability refactoring in a separate PR.